Added tag for extra assembly search path

ZimM-LostPolygon opened the pull request:
Currently, referenced assembly are only searched at InPath. I've added a `<AssemblySearchPath>` tag that specifies an additional place to search for references. There can be multiple instances of this tag, of course. Usage example:
```
<?xml version='1.0'?>
<Obfuscator>
  <Var name="InPath" value=".\Assets\SpriteSharp\Editor" />
  <Var name="OutPath" value=".\Assets\SpriteSharp\Editor" />

  <Var name="KeepPublicApi" value="true" />

  <AssemblySearchPath path=".\Library\UnityAssemblies" />
  <AssemblySearchPath path=".\Assets\SpriteSharp\Editor\3rdParty" />

  <Module file="$(AssemblyPath)\LostPolygon.SpriteSharp.dll" />
</Obfuscator>
```

lextm commented on Obfuscar/Project.cs:
```diff
@@ -56,12 +57,21 @@ private Project ()
 			}
 		}
 
+		public IEnumerable<string> AssemblySearchPaths {
```
This property does not simply use `assemblySearchPaths`, so I suggest renaming it to another name, like `AllPathsToSearch`.

ZimM-LostPolygon commented:
Renamed to AllAssemblySearchPaths, both descriptive and clear, I guess.

01.05.2017 02:30, Lex Li пишет:
>
> *@lextm* requested changes on this pull request.
>
> Thanks for the contribution. If you can revise the naming issue I 
> pointed out, I am happy to merge it.
>
> ------------------------------------------------------------------------
>
> In Obfuscar/Project.cs 
> <https://github.com/lextm/obfuscar/pull/100#discussion_r114088483>:
>
> >  			get {
>   				return vars.GetValue ("ExtraFrameworkFolders", "").Split (new char [] { Path.PathSeparator }, StringSplitOptions.RemoveEmptyEntries);
>   			}
>   		}
>   
> +		public IEnumerable<string> AssemblySearchPaths {
>
> This property does not simply use |assemblySearchPaths|, so I suggest 
> renaming it to another name, like |AllPathsToSearch|.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/lextm/obfuscar/pull/100#pullrequestreview-35529489>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AA4rJidsbnRsIQB7Zjvwn7uvI6az-9Xxks5r1SeegaJpZM4NMpMc>.
>

ZimM-LostPolygon commented:
Accidentally used the wrong branch for this PR. Closed and reopened in https://github.com/lextm/obfuscar/pull/102
